File: prob30.py
#!/usr/bin/env python

# combos = []
# for a in range(1,10):
#     for b in range(0,10):
#         for c in range(0,10):
#             for d in range(0,10):
#                 num = int(str(a)+str(b)+str(c)+str(d))
#                 numsum = a**4 + b**4 + c**4 + d**4
#                 if num == numsum:
#                     print(num)
#                     combos.append(num)
# print(sum(combos))

combos = []
numdigits = 5
maxnum = int("9"*(numdigits+1))
minnum = 2
num = minnum

while num <= maxnum:
    digits = [int(i) for i in list(str(num))]
    fourthed = sum([i**numdigits for i in digits])
    if fourthed == num:
        print(num)
        combos.append(num)
    num += 1

print(sum(combos))

File: prob30.go
package main

import (
	"fmt"
	"math"
	"strconv"
	"strings"
)

func main() {
	combos := []int64{}
	const numdigits = 5
	const minnum = 2
	var maxnum, _ = strconv.ParseInt(strings.Repeat("9", numdigits+1), 10, 64)
	var num = int64(minnum)

	//var numstr string
	var fourthed float64
	for ; num <= maxnum; num += 1 {
		//numstr = strconv.Itoa(num)
		fourthed = 0
		for _, a := range strconv.FormatInt(num, 10) {
			i, _ := strconv.Atoi(string(a))
			fourthed += math.Pow(float64(i), numdigits)
		}
		if int64(fourthed) == num {
			combos = append(combos, num)
			fmt.Println(num)
		}
	}

	var sum int64
	for _, i := range combos {
		sum += i
	}

	fmt.Println("Sum of combos:", sum)
}
